test/albumController.test.js

```diff
@@ -0,0 +1,187 @@
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Album = require('../src/models/Album');
+const { loginAsAdmin } = require('./test_helper');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('AlbumController API Tests', () => {
+  const mockAlbumData = {
+    albumName: 'Test Album',
+    artistName: 'Test Artist',
+    price: 9.99,
+    spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+  };
+
+  beforeEach(async () => {
+    await Album.deleteMany({});
+    await Album.create(mockAlbumData);
+  });
+
+  it('should test the getAllAlbums endpoint - Success Case', async () => {
+    const response = await request(app).get('/api/v1/albums');
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body.albums).toHaveLength(1);
+  });
+
+  it('should return an empty list if there are no albums in the database', async () => {
+    await Album.deleteMany({});
+    // Make a GET request to the /api/v1/albums endpoint
+    const response = await request(app).get('/api/v1/albums');
+
+    // Assert that the response status is OK (200)
+    expect(response.status).toBe(StatusCodes.OK);
+
+    // Assert that the response body is an object with "albums" and "count" keys
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body).toHaveProperty('count');
+
+    // Assert that "albums" is an empty array since there are no albums in the database
+    expect(response.body.albums).toEqual([]);
+
+    // Assert that the "count" field reflects the actual count of albums (which is 0 in this case)
+    expect(response.body.count).toBe(0);
+  });
+
+  it('should test the getSingleAlbum endpoint - Success Case', async () => {
+    const { _id: albumId } = await Album.findOne({});
+
+    const response = await request(app).get(`/api/v1/albums/${albumId}`);
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+  });
+
+  it('should test the getSingleAlbum endpoint - Error Case (Not Found)', async () => {
+    const response = await request(app).get('/api/v1/albums/invalidAlbumId');
+    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+  });
+
+  it('should test the getAlbumWithAllUsersWhoPurchasedIt endpoint - Success Case', async () => {
+    const { _id: albumId } = await Album.findOne({});
+
+    // Login as admin
+    const loginResponse = await loginAsAdmin(app);
```

Comment:
I have tried to change it but still have  errors lines 93 and 184 with limit of albums 🥹 ,commit f4d6bbbbbef4da708efad289e00c456c8968988c

Reply:
it can't find cookie 
<img width="679" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/42424ee4-0bdf-49d9-ae2a-be76e8ce7091">